storage/src/main/java/tech/pegasys/teku/storage/client/ChainHead.java

```diff
@@ -62,23 +61,25 @@ public UInt64 findCommonAncestor(final ChainHead other) {
             .minus(spec.getSlotsPerHistoricalRoot(slot));
     while (slot.isGreaterThan(minSlotWithHistoricRoot)) {
       if (getBlockRootAtSlot(slot).equals(other.getBlockRootAtSlot(slot))) {
-        return slot;
+        return new SlotAndBlockRoot(slot, getBlockRootAtSlot(slot));
```

Comment:
We shouldn't be requesting the block root multiple times as that duplicates processing. Should store `getBlockRootAtSlot(slot)` into a variable so it can be reused.